Additional partner fields

frankyma opened the pull request:
Loading: <img width="1282" alt="Screen Shot 2020-10-13 at 1 12 48 PM" src="https://user-images.githubusercontent.com/43487134/95911182-fcd30080-0d55-11eb-9290-bca212377116.png">
Partner Scene: <img width="1270" alt="Screen Shot 2020-10-13 at 1 13 37 PM" src="https://user-images.githubusercontent.com/43487134/95911188-ff355a80-0d55-11eb-973c-3f0d8d175f75.png">

frankyma commented on src/scenes/Partners/Edit/EditPartner.tsx:
```diff
@@ -0,0 +1,161 @@
+      {...props}
+    />
+  ),
+  //TODO: fix hard crash and missing initialValues
```
Max rerender crashing here, and also for some reason the initialValues aren't populating.

frankyma replied:
Fixed by Carson's fix!

glassblowerscat replied:
This isn't populating because we're giving a radio input, which takes a single string value, an array, which is what our `UpdatePartner` input wants us to use. This seems like a mismatch between the UI and the schema to me. Are users supposed to be able to submit multiple `financialReportingTypes` for a partner? Then this should be a checkbox field. Otherwise, `UpdatePartner` should take an Enum, not an array of Enums

CarsonF replied:
Yes multiple here. This only defines what's allowed to be selected in the partnership object

frankyma replied:
Got it, changed to multiple and fixed.  I noticed that in UpdatePartnership the `financialReportingType` is singular, is that intended?

CarsonF replied:
No it should be multiple. I had to point that out on the API PR too, so it must've been missed in the update.

frankyma commented on src/scenes/Partners/Detail/AddressCard.tsx:
```diff
@@ -0,0 +1,58 @@
+import {
```
Let me know if you think I should move this and `PartnerTypesCard.tsx` to the components directory, wasn't sure if this may be reusued

glassblowerscat replied:
I don't see anything else in the existing designs that resembles this, so I say we just leave it here unless it changes to more closely resemble another existing component. I suppose there's a chance we'll need it or something very similar to it again elsewhere in the app, won't see it in here, and will end up duplicating some code. But ideally any time we need to create some UI we should be looking in Figma to see if it already exist in the designs for a scene we've already implemented, and if so, we ought to find it then.

glassblowerscat commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -1,16 +1,38 @@
+import { DataButton } from '../../../components/DataButton';
+import { useDialog } from '../../../components/Dialog';
+import { useDateFormatter } from '../../../components/Formatters';
+import { PencilCircledIcon } from '../../../components/Icons';
```
This is different than the component we use in other similar scenes. From Language Engagement Detail:

![image](https://user-images.githubusercontent.com/728777/96351788-c534c300-108b-11eb-9c17-e735a6efb76e.png)

frankyma replied:
I actually copied this from Project Details:
<img width="1268" alt="Screen Shot 2020-10-18 at 11 26 08 AM" src="https://user-images.githubusercontent.com/43487134/96376591-c40ba080-1134-11eb-8531-c374ef9358ee.png">

I believe in language engagement details we have a breadcrumb to project and a link out to language because we have details scenes for those, but here we don't have a detail scene for Organization.

glassblowerscat commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +109,97 @@ export const PartnerDetail = () => {
+            >
+              <AddCircle />
+            </IconButton>
+            <Typography>Edit</Typography>
```
This seems like another place where we should maybe use the same FAB as above. I realize this reflects the design, so maybe we need to ask the design team for clarification on this.

If they want to stick with the existing design, we should make sure the word "Edit" is clickable

CarsonF replied:
Yeah we've translated the word to a tooltip so we don't have to make the word clickable with the FAB

frankyma replied:
Sorry guys not sure what the decision here is.  Keep as is (edit word non-clickable) or change it to a tooltip, something else?

CarsonF replied:
Let's not match the designs and instead make it a Fab with a tooltip. This makes it consistent with the rest of the app.

If design doesn't like this then we can talk about the reasons for us doing this (text is clickable that isn't styled as it) and possibly update across the app.

glassblowerscat commented on src/scenes/Partners/Detail/PartnerTypesCard.tsx:
```diff
@@ -0,0 +1,71 @@
+  const classes = useStyles();
+
+  return (
+    <Card className={className}>
```
Our other similar cards are clickable using a `CardActionArea`. The "Edit" button is just there for a UI signal to the user

frankyma replied:
Added `CardActionArea`

glassblowerscat commented on src/scenes/Partners/Detail/AddressCard.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  return (
+    <Card className={className}>
+      <CardContent className={classes.cardContent}>
```
This should take a `CardActionArea` as well

CarsonF reviewed (changes requested):
Empty state for pmc should start with "Enter ..." to match everything else.

---

![Screen Shot 2020-10-20 at 11 57 14 AM](https://user-images.githubusercontent.com/932566/96619465-d835d600-12cb-11eb-8a9c-d1b6347a4137.png)

These empty states look pretty poor. Can some more UI be added here?

---

Address should be a textarea. See https://github.com/SeedCompany/cord-field/blob/d2042d4f57da660f6d92d8dbbd3fc68efd2c084b/src/scenes/Users/UserForm/UserForm.tsx#L128-L134

CarsonF commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +113,98 @@ export const PartnerDetail = () => {
+                onClick={() => editPartner('globalInnovationsClient')}
               >
                 <PencilCircledIcon />
               </IconButton>
```
Use FAB. Tooltip should match aria-label.

frankyma replied:
Changed.  I haven't used many FABs before this, what's the difference between this and a regular icon button?  From what I've read looks like they're mostly for those fixed position buttons on mobile screens?

CarsonF commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +113,98 @@ export const PartnerDetail = () => {
+            </Typography>
+            {partner && (
+              <Typography variant="body2" color="textSecondary">
+                Created {formatDate(partner.createdAt)}
```
Use date time formatting for DateTime fields.

frankyma replied:
Changed.  I did hear Amy say that we don't need to show the time on alot of these, if that becomes important we can either just use the date formatter for dateTime or we'll have to change the type

CarsonF commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +113,98 @@ export const PartnerDetail = () => {
+          </Grid>
+          <Grid container spacing={3}>
+            <Grid item xs={6} className={classes.cardSection}>
+              <Typography variant="h3">Partner Types</Typography>
```
It's weird that these don't have loading states. The scene type even does ("Partner")

frankyma replied:
Added loading states, this is what the loading screen looks like now:
<img width="1314" alt="Screen Shot 2020-10-20 at 1 32 37 PM" src="https://user-images.githubusercontent.com/43487134/96641997-59946500-12da-11eb-94e9-73c94d1106df.png">

CarsonF commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +113,98 @@ export const PartnerDetail = () => {
+                label="Global Innovations Client"
+                redacted="You do not have permission to view whether this is a Global Innovations Client"
+                data={partner?.globalInnovationsClient}
+                wrap={(node) => <Grid item>{node}</Grid>}
```
You've got two grid items here. One is unnecessary.

CarsonF commented on src/scenes/Partners/Detail/PartnerDetail.tsx:
```diff
@@ -60,21 +113,98 @@ export const PartnerDetail = () => {
+                    : ''
+                }`}
+              />
+            </Grid>
```
Let's put status first, like the designs have it. Set the `empty` prop here too.

frankyma replied:
Changed

CarsonF commented on src/scenes/Partners/Detail/PartnerTypesCard.tsx:
```diff
@@ -0,0 +1,78 @@
+            {partner?.types.value.includes('Managing') && (
+              <Grid item>
+                <Typography variant="body2" color="textSecondary">
+                  Financial Reporting Type
```
```suggestion
                  Financial Reporting Types
```

frankyma commented:
> Empty state for pmc should start with "Enter ..." to match everything else.

Changed.  I got that from LanguageEngagementDetail DataButton so changed it there also

Added Empty State text to cards 
<img width="1112" alt="Screen Shot 2020-10-20 at 12 26 42 PM" src="https://user-images.githubusercontent.com/43487134/96634544-87c07780-12cf-11eb-9d67-61abd4299820.png">

> Address should be a textarea. See
> 
> https://github.com/SeedCompany/cord-field/blob/d2042d4f57da660f6d92d8dbbd3fc68efd2c084b/src/scenes/Users/UserForm/UserForm.tsx#L128-L134

Changed
![Uploading Screen Shot 2020-10-20 at 12.27.14 PM.png…]()

glassblowerscat reviewed (approved):
🕺
